src/rules/noMagicNumbersRule.ts

```diff
@@ -87,8 +154,35 @@ class NoMagicNumbersWalker extends Lint.AbstractWalker<Set<string>> {
     }
 
     private checkNumericLiteral(node: ts.Node, num: string) {
-        if (!Rule.ALLOWED_NODES.has(node.parent!.kind) && !this.options.has(num)) {
+        if (
+            /* "allow-element-access" logic */
+            !(
+                node.parent !== undefined &&
+                isElementAccessExpression(node.parent) &&
+                this.options.allowElementAccess
```

Comment:
please pull this `()` out to a local variable so it's easier to follow this logic.